src/routers/catalog.ts

```diff
@@ -1,18 +1,36 @@
+import * as Effect from "@effect/io/Effect";
+import { pipe } from "@effect/data/Function";
 import { zodiosRouter } from "@zodios/express";
 import { api } from "../model/generated/api.js";
-import { ApiError, mapCatalogServiceErrorToApiError } from "../model/types.js";
+import {
+  makeUnexpectedErrorProblem,
+  mapCatalogServiceErrorToApiError,
+} from "../model/types.js";
 import { catalogService } from "../services/CatalogService.js";
+import { db, DB } from "../repositories/db.js";
 
 const eservicesRouter = zodiosRouter(api.api);
 
-eservicesRouter.post("/eservices", async (req, res) => {
-  try {
-    await catalogService.createEService(req.body);
-    return res.status(201).end();
-  } catch (error) {
-    const errorRes: ApiError = mapCatalogServiceErrorToApiError(error);
-    return res.status(errorRes.status).json(errorRes).end();
-  }
-});
+eservicesRouter.post("/eservices", (req, res) =>
+  Effect.runPromise(
+    pipe(
+      catalogService.createEService(req.body),
+      Effect.provideService(DB, db),
+      Effect.catchAllDefect(() =>
+        Effect.sync(() => {
+          const errorRes = makeUnexpectedErrorProblem();
+          res.status(errorRes.status).json(errorRes).end();
+        })
+      ),
+      Effect.match(
+        (error) => {
+          const errorRes = mapCatalogServiceErrorToApiError(error);
+          return res.status(errorRes.status).json(errorRes).end();
+        },
```

Comment:
Qui invece la gestione degli errori noti, che vengono semplicemente mappati in maniera esaustiva (vedi sopra)